spec-tests/src/test/java/org/candlepin/spec/CloudRegistrationSpecTest.java

```diff
@@ -14,85 +14,469 @@
+
+        assertTokenType(adminClient.MAPPER, result.getToken(), TOKEN_TYPE);
+        assertThat(result)
+            .isNotNull()
+            .returns(owner.getKey(), CloudAuthenticationResultDTO::getOwnerKey)
+            .returns(null, CloudAuthenticationResultDTO::getAnonymousConsumerUuid)
+            .returns(TOKEN_TYPE, CloudAuthenticationResultDTO::getTokenType);
+    }
+
+    @Test
+    public void shouldReceiveAnonTokenForV2AuthWithExistingOwnerForCloudAccountIdAndNoEntitlement()
+        throws Exception {
+        ApiClient adminClient = ApiClients.admin();
+        OwnerDTO owner = adminClient.owners().createOwner(Owners.random());
+        adminClient.hosted().createOwner(owner);
+        String accountId = StringUtil.random("cloud-account-id-");
+        String instanceId = StringUtil.random("cloud-instance-id-");
+        String offerId = StringUtil.random("cloud-offer-");
+
+        associateProductIdToCloudOffer(adminClient, offerId, StringUtil.random("prod-"));
+        associateOwnerToCloudAccount(adminClient, accountId, owner.getKey());
+
+        String metadata = buildMetadataJson(adminClient.MAPPER, accountId, instanceId, offerId);
+
+        CloudAuthenticationResultDTO result = adminClient.cloudAuthorization()
+            .cloudAuthorizeV2(generateToken(metadata, "test-type", ""));
+
+        assertTokenType(adminClient.MAPPER, result.getToken(), ANON_TOKEN_TYPE);
+        assertThat(result)
+            .isNotNull()
+            .returns(owner.getKey(), CloudAuthenticationResultDTO::getOwnerKey)
+            .doesNotReturn(null, CloudAuthenticationResultDTO::getAnonymousConsumerUuid)
+            .returns(ANON_TOKEN_TYPE, CloudAuthenticationResultDTO::getTokenType);
+    }
+
+    @Test
+    public void shouldReceiveAnonTokenForV2AuthWithNoOwnerForCloudAccountId() throws Exception {
+        ApiClient adminClient = ApiClients.admin();
+        String accountId = StringUtil.random("cloud-account-id-");
+        String instanceId = StringUtil.random("cloud-instance-id-");
+        String offerId = StringUtil.random("cloud-offer-");
+
+        associateProductIdToCloudOffer(adminClient, offerId, StringUtil.random("prod-"));
+
+        String metadata = buildMetadataJson(adminClient.MAPPER, accountId, instanceId, offerId);
+
+        CloudAuthenticationResultDTO result = adminClient.cloudAuthorization()
+            .cloudAuthorizeV2(generateToken(metadata, "test-type", ""));
+
+        assertTokenType(adminClient.MAPPER, result.getToken(), ANON_TOKEN_TYPE);
+        assertThat(result)
+            .isNotNull()
+            .returns(null, CloudAuthenticationResultDTO::getOwnerKey)
+            .doesNotReturn(null, CloudAuthenticationResultDTO::getAnonymousConsumerUuid)
+            .returns(ANON_TOKEN_TYPE, CloudAuthenticationResultDTO::getTokenType);
+    }
+
+    @Test
+    public void shouldAllowV2AuthWithExistingEntitledAnonymousOwnerForCloudAccountId() throws Exception {
```

Comment:
This test seems identical with the first one here `shouldAllowV2AuthWithExistingEntitledOwnerForCloudAccountId`, however they do have different names. Was it by accident, or is one supposed to test with an anonymous owner and the other with a normal owner?

Reply:
On second thought it probably makes more sense to merge CANDLEPIN-646 first and then update the test

Reply:
Got CANDLEPIN-646 merged in and rebased this branch with the changes and updated the `shouldAllowV2AuthWithExistingEntitledAnonymousOwnerForCloudAccountId` test!